File: Projects/JavaScript/project-002/main.js
async function get_api() {
  const api_url_latest =
    'https://awu4j6hku3.execute-api.eu-central-1.amazonaws.com/dev/weather/latest';
  const api_url =
    'https://awu4j6hku3.execute-api.eu-central-1.amazonaws.com/dev/weather/hist?days=1&group=1';

  const response = await fetch(api_url);

  var data = await response.json();
  console.log(data);
  // getting_data(data);
}

function getting_data(data) {
  const temps = [];
  const humidity = [];
  const wind = [];
  const pressure = [];

  for (const item of data) {
    switch (item.dataType) {
      case 'temp':
        temps.push(item.value);
        break;
      case 'humidity':
        humidity.push(item.value);
        break;
      case 'wind':
        wind.push(item.value);
        break;
      case 'pressure':
        pressure.push(item.value);
        break;
      default:
        // Handle unrecognized dataType if necessary
        break;
    }
  }

  console.log('Temperatures:', temps);
  console.log('Humidity:', humidity);
  console.log('Wind:', wind);
  console.log('Pressure:', pressure);
}

get_api();  // print them ones and then with the function setinterval it printem
            // them every k milisec after the first print

// const eep = setInterval(get_api, 60000);

File: Projects/JavaScript/project-002/main.py
import requests
from datetime import datetime
from zoneinfo import ZoneInfo

crt_rom_dt = datetime.now(tz=ZoneInfo("Europe/Bucharest"))
crt_rom_str = crt_rom_dt.isoformat('T')
# just for testing

# crt_rom_str = "2023-05-05T19:04:11.985270+03:00"
url = "https://awu4j6hku3.execute-api.eu-central-1.amazonaws.com/dev/weather"
data = [
    {
        "time": crt_rom_str,
        "value": 1,
        "dataType": "temp"
    },
    {
        "time": crt_rom_str,
        "value": 27.8,
        "dataType": "humidity"
    },
    {
        "time": crt_rom_str,
        "value": 96.8,
        "dataType": "wind"
    },
    {
        "time": crt_rom_str,
        "value": 180,
        "dataType": "pressure"
    }
]

headers = {'Content-type': 'application/json',
           'x-api-key': '216e0b0d-1c08-4eb6-aa43-0cccddf9bdbf'}

r = requests.post(url, json=data, headers=headers)

print(r.status_code)
print(r.text)
